src/network/websockets/server/index.ts

```diff
@@ -53,6 +53,25 @@ export function runWebsocketServer<AuthDetails>(params: Params<AuthDetails>) {
       socket.send(JSON.stringify(obj));
     };
 
+    let cachedMessages: Array<string> = [];
+
+    const processCache = () => {
+      if (cachedMessages.length > 0) {
+        for (var msg of cachedMessages) {
+          handleMsg(msg);
+        }
+        cachedMessages = [];
+      }
+    };
+
+    const addToCache = (msg: any) => {
+      if (cachedMessages.length > 100) {
+        console.warn("Cache Size exceeded maximum allowed length");
+        socket.terminate();
+      }
+      cachedMessages.push(JSON.stringify(msg));
```

Comment:
Let’s store message objects in our cachedMessages array. Then we don’t need to do the stringify/parse dance. To achieve that, the JSON parsing should be in `socket.on('message')` rather than in handleMsg.